physicalplan/plan.go

```diff
@@ -52,7 +52,7 @@ func newOperator(expr parser.Expr, storage storage.Queryable, mint time.Time, ma
 			operator := exchange.NewConcurrent(
 				exchange.NewCancellable(
 					scan.NewVectorSelector(
-						model.NewVectorPool(stepsBatch), filter, mint, maxt, step, lookbackDelta, stepsBatch, i, numShards)), 2)
+						model.NewVectorPool(stepsBatch), filter, mint, maxt, step, lookbackDelta, e.OriginalOffset, stepsBatch, i, numShards)), 2)
```

Comment:
Can we also do the same mint/maxt adjustment for vector and matrix selectors?

Reply:
No, this offset needs to be propagated anyway since we need to change `o.currentStep` value.

Reply:
Got it 👍